overlord/snapshotstate/snapshotstate_test.go

```diff
@@ -744,6 +746,122 @@ exec /bin/tar "$@"
 	c.Check(string(out), check.Equals, "")
 }
 
+func (snapshotSuite) testSaveIntegrationTarFails(c *check.C, tarLogLines int, expectedErr string) {
+	if os.Geteuid() == 0 {
+		c.Skip("this test cannot run as root (runuser will fail)")
+	}
+	c.Assert(os.MkdirAll(dirs.SnapshotsDir, 0755), check.IsNil)
+	// sanity check: no files in snapshot dir
+	out, err := exec.Command("find", dirs.SnapshotsDir, "-type", "f").CombinedOutput()
+	c.Assert(err, check.IsNil)
+	c.Check(string(out), check.Equals, "")
+
+	homedir := filepath.Join(dirs.GlobalRootDir, "home", "a-user")
+	// mock tar so that it outputs a desired number of lines
+	tarFailScript := fmt.Sprintf(`
+export LANG=C
+for c in $(seq %d); do echo "log line $c" >&2 ; done
+exec /bin/tar "$@"
```

Comment:
(nitpick) I guess tar is /bin/tar everywhere but on the odd change it is, maybe not use the absolute path?

Reply:
We have to use the absolute path, otherwise the mocked 'tar' script would end up calling itself in a loop.